### PR TITLE
Add joining primitives tests

### PR DESCRIPTION
**Pull request by FrancoCorleone**

### CODE REVIEW

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    private static final String SEPARATOR = String.valueOf(SEPARATOR_CHAR);
+
+    private static final String EXPECTED_STRING = "1-2-3-4-5-6-7-8-9";
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8CollectorsJoining() {
+        assertThat(Arrays.stream(PRIMITIVE_ARRAY)
+                .mapToObj(String::valueOf)
+                .collect(Collectors.joining(SEPARATOR))).isEqualTo(EXPECTED_STRING);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Bring the ".isEqualTo" down to the next line.

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private static final char SEPARATOR_CHAR = '-';
+
+    private static final String SEPARATOR = String.valueOf(SEPARATOR_CHAR);
+
+    private static final String EXPECTED_STRING = "1-2-3-4-5-6-7-8-9";
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8CollectorsJoining() {
+        assertThat(Arrays.stream(PRIMITIVE_ARRAY)
+                .mapToObj(String::valueOf)
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Indentation guidelines require only two spaces over on continuation lines.

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                .collect(Collectors.joining(SEPARATOR))).isEqualTo(EXPECTED_STRING);
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8StringJoiner() {
+        StringJoiner stringJoiner = new StringJoiner(SEPARATOR);
+        Arrays.stream(PRIMITIVE_ARRAY)
+                .mapToObj(String::valueOf)
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: indentation

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 1 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Done

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 2 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: You are right, I had some issues with my formatter

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 3 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Same

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    private char separatorChar = '-';
+
+    private String separator = String.valueOf(separatorChar);
+
+    private String expectedIntString = "1-2-3-4-5-6-7-8-9";
+
+    private String expectedCharString = "a-b-c-d-e-f";
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8CollectorsJoining() {
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Let's have separate test methods for the int array vs the char array for all the examples. The test names could be like givenIntArray_... and givenCharArray_...

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          .collect(Collectors.joining(separator)))
+          .isEqualTo(expectedCharString);
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8StringJoiner() {
+        StringJoiner intStringJoiner = new StringJoiner(separator);
+        Arrays.stream(intArray)
+          .mapToObj(String::valueOf)
+          .forEach(intStringJoiner::add);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Remember to include a blank line before the beginning of the assertions.

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        assertThat(intStringJoiner.toString()).isEqualTo(expectedIntString);
+
+        StringJoiner charStringJoiner = new StringJoiner(separator);
+        CharBuffer.wrap(charArray).chars()
+          .mapToObj(intChar -> String.valueOf((char) intChar))
+          .forEach(charStringJoiner::add);
+        assertThat(charStringJoiner.toString()).isEqualTo(expectedCharString);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: insert blank line

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_GuavaJoiner() {
+        assertThat(Joiner.on(separator).join(Ints.asList(intArray))).isEqualTo(expectedIntString);
+
+        assertThat(Joiner.on(separator).join(Chars.asList(charArray))).isEqualTo(expectedCharString);
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java7StringBuilder() {
+        assertThat(joinWithStringBuilder(intArray, separator)).isEqualTo(expectedIntString);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: go ahead and show a test method for the char array, even though it is the same

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 4 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Ok, I'll split each test into two

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 5 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Done

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 6 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Done

File: java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java (hunk 7 above)
Comment (FrancoCorleone) [reply java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java]: Done

**Pull request merged**